file/writable_file_writer.cc

```diff
@@ -226,6 +226,9 @@ IOStatus WritableFileWriter::Pad(const IOOptions& opts,
   pending_sync_ = true;
   uint64_t cur_size = filesize_.load(std::memory_order_acquire);
   filesize_.store(cur_size + pad_bytes, std::memory_order_release);
+
+  Slice data(buf_.BufferStart() + pad_start, pad_bytes);
+  UpdateFileChecksum(data);
```

Comment:
Should this be in the `while` loop above?

Reply:
Do you mean UpdateFileChecksum(data) as we pad?

Reply:
Yeah, I mean theoretically `pad_bytes` can be `> buf_.Capacity()`, right?

Reply:
Fixed

Reply:
I did something like below but it has 3% regression
```
  size_t current_pad_start = pad_start;
  while (left) {
    size_t append_bytes = std::min(cap, left);
    buf_.PadWith(append_bytes, 0);
    left -= append_bytes;
    if (left > 0) {
      IOStatus s = Flush(io_options);
      if (!s.ok()) {
        set_seen_error();
        return s;
      }
    }
    Slice data(buf_.BufferStart() + current_pad_start,
               append_bytes);
    UpdateFileChecksum(data);
    cap = buf_.Capacity() - buf_.CurrentSize();
    current_pad_start += append_bytes;
  }
 ```
Pre-PR:
fillseq [AVG    300 runs] : 420613 (± 3865) ops/sec;   46.5 (± 0.4) MB/sec
Post-PR: regress 1%
fillseq [AVG    300 runs] : 416331 (± 3783) ops/sec;   46.1 (± 0.4) MB/sec
 
Something like this gives less than 1% regression. It considered the actual appended bytes instead of `pad_bytes` outside of the loop. The PR currently contains this solution.
 ```
 while (left) {
    ...
 }
 pending_sync_ = true;
  uint64_t cur_size = filesize_.load(std::memory_order_acquire);
  filesize_.store(cur_size + pad_bytes, std::memory_order_release);
  Slice data(buf_.BufferStart() + pad_start,
               buf_.CurrentSize() - pad_start);
  UpdateFileChecksum(data);
  if (perform_data_verification_) {
  ...
```
Pre-PR:
fillseq [AVG    300 runs] : 419372 (± 3950) ops/sec;   46.4 (± 0.4) MB/sec
Post-PR: regress 0.5%
fillseq [AVG    300 runs] : 417097 (± 3875) ops/sec;   46.1 (± 0.4) MB/sec

Reply:
Maybe I'm missing something. On line 234 below, its `crc32c::Extend(buffered_data_crc32c_checksum_, buf_.BufferStart() + pad_start, pad_bytes)`. Should we also do `(buf_.BufferStart() + pad_start, pad_bytes)`? What the PR currently has is `(buf_.BufferStart() + pad_start, buf_.CurrentSize() - pad_start)`, which may not necessarily equal `pad_bytes` if `buf_.CurrentSize()` was reset in the while loop above by `Flush()`.

Reply:
Oh sorry I overlooked the flush part. Fixed it. Also, for your original concern of "pad_bytes can be > buf_.Capacity()", I calculated and used the actual appended bytes for both of these two places. No obvious regression is shown (actually a slight improvment)

Pre-PR:
fillseq [AVG    300 runs] : 422857 (± 3942) ops/sec;   46.8 (± 0.4) MB/sec
Post-PR: 
fillseq [AVG    300 runs] : 424707 (± 3799) ops/sec;   47.0 (± 0.4) MB/sec

Reply:
> Should this be in the while loop above?

+1, perf consequences are OK with me